src/effects/VST3/VST3Effect.cpp

```diff
@@ -0,0 +1,787 @@
+{
+   //Dummy
+   mSetup.maxSamplesPerBlock = static_cast<Steinberg::int32>(maxBlockSize);
+   return mSetup.maxSamplesPerBlock;
+}
+
+size_t VST3Effect::GetBlockSize() const
+{
+   return mSetup.maxSamplesPerBlock;
+}
+
+sampleCount VST3Effect::GetLatency()
+{
+   using namespace Steinberg;
+
+   auto latency { 0u };
+   
+   for(auto& realtimeProcessor : mRealtimeGroupProcessors)
+   {
+      Vst::IAudioProcessor* audioProcessor;
+      if(realtimeProcessor->mEffectComponent->queryInterface(Vst::IAudioProcessor::iid, (void**)&audioProcessor) == kResultOk)
+         latency = std::max(
+            latency,
+            static_cast<decltype(latency)>(audioProcessor->getLatencySamples())
+         );
+   }
+
+   return latency;
+}
+
+size_t VST3Effect::GetTailSize()
+{
+   using namespace Steinberg;
+   Vst::IAudioProcessor* audioProcessor;
+   if(mEffectComponent->queryInterface(Vst::IAudioProcessor::iid, reinterpret_cast<void**>(&audioProcessor)) == kResultOk)
+      return audioProcessor->getTailSamples();
+   
+   return 0;
+}
+
+bool VST3Effect::ProcessInitialize(sampleCount, ChannelNames)
+{
+   using namespace Steinberg;
+
+   Vst::IAudioProcessor* audioProcessor;
+   if(mEffectComponent->queryInterface(Vst::IAudioProcessor::iid, reinterpret_cast<void**>(&audioProcessor)) == kResultOk)
+   {
+      if(audioProcessor->canProcessSampleSize(mSetup.symbolicSampleSize) ==kResultTrue &&
+         audioProcessor->setupProcessing(mSetup) == kResultOk)
+      {
+         if(mEffectComponent->setActive(true) == kResultOk)
+         {
+            SyncParameters();
+            audioProcessor->setProcessing(true);
+            return true;
+         }
+      }
+   }
+   return false;
+}
+
+bool VST3Effect::ProcessFinalize()
+{
+   using namespace Steinberg;
+   Vst::IAudioProcessor* audioProcessor;
+   if(mEffectComponent->queryInterface(Vst::IAudioProcessor::iid, reinterpret_cast<void**>(&audioProcessor)) == kResultOk)
+      audioProcessor->setProcessing(false);
+
+   return mEffectComponent->setActive(false) == Steinberg::kResultOk;
+}
+
+namespace
+{
+   size_t VST3ProcessBlock(
+      Steinberg::Vst::IComponent* effect,
+      const Steinberg::Vst::ProcessSetup& setup,
+      float** inBlock,
+      float** outBlock,
+      size_t blockLen,
+      Steinberg::Vst::IParameterChanges* inputParameterChanges)
+   {
+      using namespace Steinberg;
+   Vst::IAudioProcessor* audioProcessor;
+   if(effect->queryInterface(Vst::IAudioProcessor::iid, reinterpret_cast<void**>(&audioProcessor)) == kResultOk)
+   {
+      Vst::ProcessData data;
+      data.processMode = setup.processMode;
+      data.symbolicSampleSize = setup.symbolicSampleSize;
+      data.inputParameterChanges = inputParameterChanges;
+
+      static_assert(std::numeric_limits<decltype(blockLen)>::max()
+         >= std::numeric_limits<decltype(data.numSamples)>::max());
+
+      data.numSamples = static_cast<decltype(data.numSamples)>(std::min(
+         blockLen, 
```

Comment:
Maybe blockLen should be cast from an unsigned to a signed type before you take a minimum.  Could the other argument to min be a negative that indicates an invalid value?